packages/migrate/src/utils/getDatabaseVersionSafe.ts

```diff
@@ -1,15 +1,17 @@
 import Debug from '@prisma/debug'
+import { MigrateTypes } from '@prisma/internals'
 
 import { SchemaEngine } from '../SchemaEngine'
-import type { EngineArgs } from '../types'
 
 const debug = Debug('prisma:cli')
 
 /**
  * Retrieve the database version from the given schema or url.
  * This function never throws, and was introduced to prevent circular dependencies in `@prisma/internals`.
  */
-export async function getDatabaseVersionSafe(args: EngineArgs.GetDatabaseVersionParams): Promise<string | undefined> {
+export async function getDatabaseVersionSafe(
+  args: MigrateTypes.GetDatabaseVersionParams | undefined,
```

Comment:
Why the `undefined` here?

Reply:
`| undefined` was a part of `EngineArgs.GetDatabaseVersionParams` definition before already.
Since that does not quite agree with what `schema-engine` expects, i've changed the definition and so we had to add `| undefined` to a couple of signatures where we actually pass undefined values every once in a while